errbot/plugin_manager.py

```diff
@@ -272,7 +272,18 @@ def activate_plugin_with_version_check(self, plugin_info: PluginInfo) -> BotPlug
             log.error('%s failed errbot version check.', name)
             return None
 
+        # Activate dependencies of this plugin first.
+        try:
+            depends_on = [dep_name.strip() for dep_name in plugin_info.details.get("Core", "DependsOn").split(',')]
+            for dep_name in depends_on:
+                if dep_name not in self.get_all_active_plugin_names():
+                    log.debug('%s depends on %s and %s is not activated. Activating it ...', name, dep_name, dep_name)
+                    self.activate_plugin(dep_name)
+        except NoOptionError:
+            depends_on = []
```

Comment:
`activate_plugin_with_version_check` is starting to get pretty long. Maybe move this section of code into a separate function to keep it a little more readable?